apps/ai/src/app/api/agent/route.ts

```diff
@@ -0,0 +1,50 @@
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+
+import agent from "../../../utils/agent";
+import { context } from "../../../utils/context";
+import sendEmail from "../../../utils/sendEmail";
+import { verifyParseKey } from "../../../utils/verifyParseKey";
+
+/**
+ * Launches a LangChain agent to process an incoming email,
+ * then sends the response to the user.
+ */
+export const POST = async (request: NextRequest) => {
+  const verified = verifyParseKey(request.url);
+
+  if (!verified) {
+    return new NextResponse("Unauthorized", { status: 401 });
+  }
+
+  const json = await request.json();
+
+  const { context: _context, message, subject, user, replyTo } = json;
+
+  if ((!message && !subject) || !user) {
+    return new NextResponse("Missing fields", { status: 400 });
+  }
+
+  // Persist context for agent tools
+  context.apiKey = _context.apiKey;
+  context.userId = _context.userId;
```

Comment:
Wouldn't this cause context mixups if a lambda server multiple requests per invocation? 

I've seen this happen before. When we have multiple requests per second, there's the chance that a previous context gets overwritten by a newer request and request A gets data from request B and so on.

Reply:
Some more context: https://adamtuttle.codes/blog/2022/aws-lambda-safe-global-variables/

Reply:
thanks so much @zomars for this awesome review and for finding this. Working on it now.